File: backend/push-service/src/main/java/com/example/pushservice/fcm/entity/FcmToken.java
package com.example.pushservice.fcm.entity;

import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class FcmToken {
    @Id
    @Column(name = "token_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String userId;

    private String fcmToken;

    public FcmToken(String userId, String fcmToken) {
        this.userId = userId;
        this.fcmToken = fcmToken;
    }
}

File: backend/push-service/src/main/kotlin/com/example/pushservice/fcm/entity/FcmToken.kt
package com.example.pushservice.fcm.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class FcmToken(
    var userId: String? = null,
    var fcmToken: String? = null,
) {
    @Id
    @Column(name = "token_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
}
